tests/models/nucleus_x/test_modeling_nucleus_x.py

```diff
@@ -131,6 +131,7 @@ def get_config(self):
             decoder_layers=self.num_hidden_layers,
             is_decoder=False,
             decoder_retention_heads=self.num_attention_heads,
+            use_cache=False,
```

Comment:
When `use_cache=True`, `NucleusXMultiScaleRetention.parallel_forward` will be less efficient. This is because `use_cache=True` makes the `NucleusXMultiScaleRetention.parallel_forward` to compute `past_key_values`, which incurs another `O(T^2)` computations.

Only when we want to do recurrent forward following the parallel forward (e.g. during generation, we compute the prompt in parallel, but generate with recurrent mode), `use_cache=True` should be set.

Reply:
cc @gante to this bit - is our generation code ready to handle networks with multiple forward modes?